apps/bluetooth/js/modules/pair_manager_v1.js

```diff
@@ -0,0 +1,274 @@
+/* -*- Mode: js; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- /
+/* vim: set shiftwidth=2 tabstop=2 autoindent cindent expandtab: */
+
+define(function(require) {
+  'use strict';
+
+  var PairExpiredDialog = require('views/pair_expired_dialog');
+  var BluetoothHelper = require('shared/bluetooth_helper');
+
+  var _ = window.navigator.mozL10n.get;
+  var _debug = false;
+  var Debug = function() {};
+  if (_debug) {
+    Debug = function btam_debug(msg) {
+      console.log('--> [PairManagerV1]: ' + msg);
+    };
+  }
+
+  /*
+   * PairManager is responsible for:
+   *   1. Handling system message 'bluetooth-pairing-request' while there is an
+   *      incoming/outgoing pairing request.
+   *   2. handling system message 'bluetooth-cancel' while some remote devices
+   *      request for canceling an overdue pairing request. The reason could be
+   *      cancel from remote devices, timeout, or other..
+   */
+  var PairManager = {
+    init: function() {
+      // require mozBluetooth from BluetoothHelper
+      this.bluetoothHelper = BluetoothHelper();
+
+      navigator.mozSetMessageHandler('bluetooth-pairing-request',
+        this.onRequestPairing.bind(this)
+      );
+
+      navigator.mozSetMessageHandler('bluetooth-cancel',
+        this.onBluetoothCancel.bind(this)
+      );
+
+      // Observe screen lockscreen locked/unlocked state for show pending
+      // pairing request immediately.
+      navigator.mozSettings.addObserver('lockscreen.locked',
+        function gotScreenLockedChanged(event) {
+          this.showPendingPairing(event.settingValue);
+      }.bind(this));
+
+      // Observe Bluetooth ondisabled event from hardware side.
+      // Then, close pairing dialog immediately.
+      navigator.mozBluetooth.ondisabled = this.onBluetoothDisabled.bind(this);
+
+      // Observe Bluetooth enabled state for close pairing dialog immediately.
+      navigator.mozSettings.addObserver('bluetooth.enabled',
+        function gotBluetoothEnabledChanged(event) {
+          if (!event.settingValue) {
+            this.onBluetoothDisabled();
+          }
+      }.bind(this));
+    },
+
+    onRequestPairing: function(pairingInfo) {
```

Comment:
I have revised the function to be internal in `pair_manager.js`. I wouldn't like to suggestion to modify the old module. It might be created regression. This script will be ignore while we're running on Bluetooth v2.0 API. :)